api/leaderboard.go

```diff
@@ -93,7 +93,7 @@ func UpsertMemberScoreHandler(app *App) func(c echo.Context) error {
 
 		var member *leaderboard.Member
 		err = WithSegment("Model", c, func() error {
-			l := leaderboard.NewLeaderboard(app.RedisClient, leaderboardID, 0, lg)
+			l := leaderboard.NewLeaderboard(app.RedisClient.Trace(c.StdContext()), leaderboardID, 0, lg)
```

Comment:
tracing manually is really needed? ddog is also working this way?

Reply:
Yes, although in a slightly different manner. The Redis client has to know which is the current context to be able to link the spans, and the Go way of doing this is passing it on explicitly.

Reply:
@felipejfc take a look in this example usage: https://github.com/DataDog/dd-trace-go/blob/master/contrib/go-redis/redis/example_test.go#L14 you also need to pass the context explicitly.

Reply:
the beauty in the wrapper is that you don't need to write Trace on every call to Redis